x-pack/plugins/apm/server/plugin.ts

```diff
@@ -61,28 +59,26 @@ export class APMPlugin implements Plugin<APMPluginContract> {
       createApmApi().init(core, { config$: mergedConfig$, logger, __LEGACY });
     });
 
-    await new Promise(resolve => {
-      mergedConfig$.subscribe(async config => {
-        this.currentConfig = config;
-        await createApmAgentConfigurationIndex({
-          esClient: core.elasticsearch.dataClient,
-          config
-        });
-        resolve();
-      });
+    const currentConfig = await mergedConfig$.pipe(take(1)).toPromise();
+
+    // create agent configuration index without blocking setup lifecycle
+    createApmAgentConfigurationIndex({
```

Comment:
This is the primary change: we no longer block and wait for `createApmAgentConfigurationIndex` to complete